SOS/SERV/sos/sos_program.c

```diff
@@ -0,0 +1,621 @@
+
+                // save task in DB
+                uint8_t_ uint8_new_task_db_index = gl_uint8_number_of_tasks_added;
+                gl_arr_ptr_str_task[uint8_new_task_db_index] = ptr_str_task;
+                gl_uint8_number_of_tasks_added++;
+                // sort DB tasks according to task priority
+                sos_sort_database(uint8_new_task_db_index);
+
+                enu_sos_status_retval = SOS_STATUS_SUCCESS;
+            }
+            else
+            {
+                /* Do Nothing */
+            }
+        }
+    }
+
+    return enu_sos_status_retval;
+}//Line 365
+
+/**
+ *	@author				                    :	Hossam Elwahsh - https://github.com/HossamElwahsh
+ *
+ *	@brief		                            :	Deletes a task from DB
+ *  @param[in,out]  uint8_task_id 	        :   Task ID to delete from DB
+ *
+ *  @Return     SOS_STATUS_SUCCESS		    :	Success,    Task deleted successfully
+ *              SOS_STATUS_INVALID_STATE    :   Failed,     SOS Invalid State (uninitialized)
+ *              SOS_STATUS_INVALID_TASK_ID  :   Failed,     Task ID not found in DB
+ */
+enu_sos_status_t_ sos_delete_task(uint8_t_ uint8_task_id)
+{//Line 378
+
+    enu_sos_status_t_ enu_sos_status_retval = SOS_STATUS_SUCCESS;
+
+    // SOS System State Check
+    if(gl_enu_sos_scheduler_state == SOS_SCHEDULER_UNINITIALIZED)
+    {
+        enu_sos_status_retval = SOS_STATUS_INVALID_STATE;
+    }
+    else
+    {
+        // search for task ID in DB
+        str_sos_task_t_ *ptr_str_sos_task_to_delete = NULL;
+        uint8_t_ uint8_task_index_in_db = 0;
+        enu_sos_status_retval = sos_find_task(uint8_task_id, &ptr_str_sos_task_to_delete, &uint8_task_index_in_db);
+
+        if(
+                SOS_STATUS_SUCCESS == enu_sos_status_retval      &&
+                NULL_PTR != ptr_str_sos_task_to_delete
+        ) // task found
+        {
+            /* Task Found - Delete it */
+            //ptr_str_sos_task_to_delete = NULL_PTR;
+			gl_arr_ptr_str_task[uint8_task_index_in_db] = NULL_PTR;
+            sos_sort_database(uint8_task_index_in_db);
+            enu_sos_status_retval = SOS_STATUS_SUCCESS;
+        }
+        else
+        {
+            /* Task not found */
+            enu_sos_status_retval = SOS_STATUS_INVALID_TASK_ID;
+        }
+    }
+
+    return enu_sos_status_retval;
+}//Line 500
+
+/**
+ *	@author				                    :	Hossam Elwahsh - https://github.com/HossamElwahsh
+ *
+ *	@brief		                            :	Modifies a task
+ *  @param[in,out]  uint8_task_id 	        :   Task ID to delete from DB
+ *
+ *  @Return     SOS_STATUS_SUCCESS		    :	Success,    Task deleted successfully
+ *              SOS_STATUS_INVALID_STATE    :   Failed,     SOS Invalid State (uninitialized)
+ *              SOS_STATUS_INVALID_ARGS     :   Failed,     Invalid Arguments Given
+ *              SOS_STATUS_INVALID_TASK_ID  :   Failed,     Task ID not found in DB
+ */
+enu_sos_status_t_ sos_modify_task(uint8_t_ uint8_task_id)

```

Comment:
Where is the parameters to modify?
This API Only Sort the Database. While you should modify the task with the new parameters, Then Sort the database

Reply:
@MinaRaoufAziz We have discussed this in the delivery meeting, as the modification is done on the task structure pointer itself directly through the application and it was redundant to pass the data again as they will be automatically modified inside our database,

we use the modify to re-sort the DB as our sorting algorithm only sorts the modified/changed task; it does not re-sort the whole DB.

Reply:
what we can do, that we add extra checks on the modified data and delete the task from our DB if the modification was bad ?